muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/base/material-loader/fields/audio-field/index.tsx

```diff
@@ -424,9 +435,23 @@ export default class AudioField extends React.Component<AudioFieldProps, AudioFi
     //if elements is disabled
     let ElementDisabledState = this.props.readOnly ? "material-page__taskfield-disabled" : "";
 
+    let fieldSavedStateClass = "";
+    if (this.state.fieldSavedState === "ERROR") {
+      fieldSavedStateClass = "state-ERROR";
+    } else if (this.state.fieldSavedState === "SAVING") {
+      fieldSavedStateClass = "state-SAVING";
+    } else if (this.state.fieldSavedState === "SAVED") {
+      fieldSavedStateClass = "state-SAVED";
+    }
```

Comment:
These line of codes are used many places. Could consider making one global or at least one level higher function to return that class value as string. Example function could looke like:
```ts
createFieldSavedStateClass = (state: FieldStateStatus) => {
let fieldSavedStateClass = "";
    if (this.state.fieldSavedState === "ERROR") {
      fieldSavedStateClass = "state-ERROR";
    } else if (this.state.fieldSavedState === "SAVING") {
      fieldSavedStateClass = "state-SAVING";
    } else if (this.state.fieldSavedState === "SAVED") {
      fieldSavedStateClass = "state-SAVED";
    }

return fieldSavedStateClass 
}
```
Switch case would work too. And then just:

```ts
let fieldSavedStateClass = createFieldSavedStateClass(this.state.fieldSavedState)
```